src/ATen/ExpandUtils.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include "ATen/Tensor.h"
+#include <sstream>
+
+namespace at {
+
+std::tuple<Tensor> expand_inplace(const Tensor &tensor, const Tensor &to_expand) {
```

Comment:
If these functions are defined in the header, I think they should be marked inline because of the "one definition rule".